pkgs/by-name/gr/grayjay/package.nix

```diff
@@ -0,0 +1,117 @@
+{
+  buildDotnetModule,
+  fetchgit,
+  dotnetCorePackages,
+  pkgs,
+  buildNpmPackage,
+  lib,
+}:
+let
+  version = "3";
+  src = fetchgit {
+    url = "https://gitlab.futo.org/videostreaming/Grayjay.Desktop";
+    rev = "refs/tags/${version}";
+    hash = "sha256-gI1M/rDTSH16eeknHJfLA2iawoA5iGTKHfwUW4/lvv8=";
+    fetchSubmodules = true;
+    fetchLFS = true;
+  };
+  frontend = buildNpmPackage {
+    name = "grayjay-frontend";
+    inherit version;
+
+    src = "${src}/Grayjay.Desktop.Web";
+
+    npmBuildScript = "build";
+    npmDepsHash = "sha256-pTEbMSAJwTY6ZRriPWfBFnRHSYufSsD0d+hWGz35xFM=";
+
+    installPhase = ''
+      cp -r dist/ $out
+    '';
+  };
+in
+buildDotnetModule {
+  pname = "grayjay";
+  version = "3";
+
+  inherit src;
+
+  projectFile = [
+    "Grayjay.ClientServer/Grayjay.ClientServer.csproj"
+    "Grayjay.Engine/Grayjay.Engine/Grayjay.Engine.csproj"
+    "Grayjay.Desktop.CEF/Grayjay.Desktop.CEF.csproj"
+    "FUTO.MDNS/FUTO.MDNS/FUTO.MDNS.csproj"
+    "JustCef/DotCef.csproj"
+  ];
+
+  testProjectFile = [
+    "Grayjay.Desktop.Tests/Grayjay.Desktop.Tests.csproj"
+    "Grayjay.Engine/Grayjay.Engine.Tests/Grayjay.Engine.Tests.csproj"
+  ];
+
+  nugetDeps = ./deps.json;
+
+  dotnet-runtime = dotnetCorePackages.aspnetcore_8_0;
+
+  executables = [ "Grayjay" ];
+
+  preBuild = ''
+    rm -r Grayjay.ClientServer/wwwroot/web
+    cp -r ${frontend} Grayjay.ClientServer/wwwroot/web'';
+
+  postInstall = ''
+    chmod +x $out/lib/grayjay/cef/dotcefnative
+    rm $out/lib/grayjay/Portable
+    ln -s /tmp/grayjay-launch $out/lib/grayjay/launch
+    ln -s /tmp/grayjay-cef-launch $out/lib/grayjay/cef/launch
+  '';
+
+  makeWrapperArgs = [
+    "--chdir"
+    "${placeholder "out"}/lib/grayjay"
+  ];
+
+  runtimeDeps = with pkgs; [
+    libz
+    icu
+    openssl # For updater
+
+    xorg.libX11
+    xorg.libXcomposite
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXrandr
+    xorg.libxcb
+
+    gtk3
+    glib
+    nss
+    nspr
+    dbus
+    atk
+    cups
+    libdrm
+    expat
+    libxkbcommon
+    pango
+    cairo
+    udev
+    alsa-lib
+    mesa
+    libGL
+    libsecret
+  ];
+
+  meta = with lib; {
+    description = "Multi-platform media application that allows you to watch content from multiple platforms in a single application";
+    longDescription = ''Grayjay is a multi-platform media application that allows you to watch content from multiple platforms in a single application. Using an extendable plugin system developers can make new integrations with additional platforms. Plugins are cross-compatible between Android and Desktop.'';
+    homepage = "https://grayjay.app/desktop/";
+    license = {
+      fullName = "FUTO Source First License 1.1";
+      url = "https://gitlab.futo.org/videostreaming/Grayjay.Desktop/-/blob/master/LICENSE.md";
+    };
```

Comment:
Should be added to the license list. Also missing `free = false` and `redistributable = true`.

https://github.com/NixOS/nixpkgs/blob/master/lib/licenses.nix